src/model.ts

```diff
@@ -0,0 +1,189 @@
+      /**
+       * Delete a model.
+       *
+       * @see [Models: delete API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/delete}
+       *
+       * @method Model#delete
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.delete((err, apiResponse) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [apiResponse] = await model.delete();
+       */
+      delete: true,
+
+      /**
+       * Check if the model exists.
+       *
+       * @method Model#exists
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {boolean} callback.exists Whether the model exists or not.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.exists((err, exists) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [exists] = await model.exists();
+       */
+      exists: true,
+
+      /**
+       * Get a model if it exists.
+       *
+       * @see [Models: get API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get}
+       *
+       * @method Model#get:
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {Model} callback.model The {@link Model}.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.get((err, model2, apiResponse) => {
+       *   // `model.metadata` has been populated.
+       * });
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [model2, apiResponse] = await model.get();
+       */
+      get: true,
+
+      /**
+       * Return the metadata associated with the Model.
```

Comment:
Sometimes it says "Model" and others "model." I guess it should always be "model" unless we're talking about a @google-cloud/bigquery "Model" object.